src/plugins/unified_search/public/saved_query_management/saved_query_management_list.tsx

```diff
@@ -112,41 +179,100 @@ const itemLabel = (attributes: SavedQueryAttributes) => {
   return label;
 };
 
-export function SavedQueryManagementList({
+const noSavedQueriesDescriptionText = [
+  i18n.translate('unifiedSearch.search.searchBar.savedQueryNoSavedQueriesText', {
+    defaultMessage: 'No saved queries.',
+  }),
+  i18n.translate('unifiedSearch.search.searchBar.savedQueryDescriptionText', {
+    defaultMessage: 'Save query text and filters that you want to use again.',
+  }),
+].join(' ');
+
+const savedQueryMultipleNamespacesDeleteWarning = i18n.translate(
+  'unifiedSearch.search.searchBar.savedQueryMultipleNamespacesDeleteWarning',
+  {
+    defaultMessage: `This saved query is shared in multiple spaces. When you delete it, you remove it from every space it is shared in. You can't undo this action.`,
+  }
+);
+
+const SAVED_QUERY_PAGE_SIZE = 5;
+const SAVED_QUERY_SEARCH_DEBOUNCE = 500;
+
+export const SavedQueryManagementList = ({
   showSaveQuery,
   loadedSavedQuery,
+  savedQueryService,
+  queryBarMenuRef,
   onLoad,
   onClearSavedQuery,
-  savedQueryService,
   onClose,
-  hasFiltersOrQuery,
-}: SavedQueryManagementListProps) {
-  const kibana = useKibana<IUnifiedSearchPluginServices>();
-  const [savedQueries, setSavedQueries] = useState([] as SavedQuery[]);
+}: SavedQueryManagementListProps) => {
+  const { uiSettings } = useKibana<IUnifiedSearchPluginServices>().services;
+  const [searchTerm, setSearchTerm] = useState('');
+  const [currentPageNumber, setCurrentPageNumber] = useState(0);
+  const [totalQueryCount, setTotalQueryCount] = useState(0);
+  const [currentPageQueries, setCurrentPageQueries] = useState<SavedQuery[]>([]);
+  const [isInitializing, setIsInitializing] = useState(true);
+  const currentPageFetchId = useRef(0);
+  const selectableRef = useRef<EuiSelectable | null>(null);
   const [selectedSavedQuery, setSelectedSavedQuery] = useState(loadedSavedQuery);
-  const [toBeDeletedSavedQuery, setToBeDeletedSavedQuery] = useState(null as SavedQuery | null);
+  const [toBeDeletedSavedQuery, setToBeDeletedSavedQuery] = useState<SavedQuery | null>(null);
   const [showDeletionConfirmationModal, setShowDeletionConfirmationModal] = useState(false);
-  const cancelPendingListingRequest = useRef<() => void>(() => {});
-  const { uiSettings, http, application } = kibana.services;
   const format = uiSettings.get('dateFormat');
 
-  useEffect(() => {
-    const fetchCountAndSavedQueries = async () => {
-      cancelPendingListingRequest.current();
-      let requestGotCancelled = false;
-      cancelPendingListingRequest.current = () => {
-        requestGotCancelled = true;
-      };
+  const debouncedSetSearchTerm = useMemo(() => {
+    return debounce((newSearchTerm: string) => {
+      setSearchTerm((currentSearchTerm) => {
```

Comment:
The does not seem to work as I would expect it:

- in my case I had 2 saved queries: "test1" and "test2"
- with "1" as search term it still shows both of them <= expected to find "test1"
- with "2" as search term it does not find any <= expected to find "test2"
- sorting changes when searched <= expected to have the list sorted the same way
- selection gets reset when searched <= expected the selection to stay

![Jan-24-2024 11-25-19](https://github.com/elastic/kibana/assets/1415710/bb3c9e73-42a0-4696-9828-00e2d22d72af)